File: TDT4120/ratatosk2.py
from sys import stdin

funksjon = stdin.readline().strip()
antall_noder = int(stdin.readline())

nodes = [-1]*antall_noder
start = int(stdin.readline())
start = 0
ratatosk = int(stdin.readline())
for linje in stdin:
    tall = linje.split()
    rot = int(tall.pop(0))
    for barn in tall:
        nodes[int(barn)] = rot
def rec(node):
    if node == -1:
        return 0
    else:
        return rec(nodes[node]) + 1
    
print rec(ratatosk) - 1

File: TDT4120/ratatosk2.c
#include<stdio.h>

struct node
{
    struct node * childs[10];
    int ratatosk;
};

void main()
{   
    char type[4];
    int nNodes;
    struct node nodes[1000];
    int i;
    scanf("%s", type);
    scanf("%d", &nNodes);
    for (i = 0; i <= nNodes; i++)
    {
        struct node newNode;
        newNode.ratatosk = 0;
        nodes[i] = newNode;
    }
    int startNode;
    int rataNode;
    scanf("%d", &startNode);
    scanf("%d", &rataNode);
    nodes[rataNode].ratatosk = 1;
    printf("%d\n", nNodes);
}
